src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/DefaultRazorHoverInfoService.cs

```diff
@@ -49,13 +50,18 @@ internal class DefaultRazorHoverInfoService : RazorHoverInfoService
             _htmlFactsService = htmlFactsService;
         }
 
-        public override HoverModel GetHoverInfo(RazorCodeDocument codeDocument, SourceLocation location)
+        public override HoverModel GetHoverInfo(RazorCodeDocument codeDocument, SourceLocation location, HoverCapability capability)
```

Comment:
I'm not super crazy about passing in the HoverCapability in this manner. I feel like @NTaylorMullen and I had discussed a similar problem in the past but I wasn't able to find that or the code relating to it. If anyone has suggestions for a better pattern I'm all ears.

Reply:
You remember correctly! 😄

Instead of passing it in you can pull it from the language server client settings https://github.com/dotnet/aspnetcore-tooling/blob/3acca07e331dabbc83e85e49d675c3aae1e00b75/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DefaultTagHelperCompletionService.cs#L224